### PR TITLE
auto-drecrement badge count before next poll

### PR DESCRIPTION
**Pull request by theoschao**
badge count is only updated every 60 seconds.
this is a quick hack to get it to update as you open notifications.

### CODE REVIEW

File: source/background.js [hunk 1]
```diff
@@ -116,6 +117,23 @@ function handleConnectionStatus(event) {
 window.addEventListener('online', handleConnectionStatus);
 window.addEventListener('offline', handleConnectionStatus);
 
+function handlePageShow(tabId, changeInfo, tab) {
+	if (changeInfo.status === 'complete') {
+		if (tab.url.slice(0, 19) === 'https://github.com/') {
```
Comment (sindresorhus) [inline source/background.js:122]: Use proper URL parsing with `new URL()`.

File: source/background.js [hunk 2]
```diff
@@ -116,6 +117,23 @@ function handleConnectionStatus(event) {
...
+			let pullId = tab.url.split('/').slice(-4).join('/')
+			let notificationState = localStore.get(pullId);
+			notificationState.then(state => {
```
Comment (sindresorhus) [inline source/background.js:125]: Use async/await.

File: source/lib/api.js [hunk 3]
```diff
@@ -78,7 +79,7 @@ export const getNotificationResponse = async (maxItems = 100) => {
 			per_page: maxItems // eslint-disable-line camelcase
 		});
 	}
-
+
```
Comment (sindresorhus) [inline source/lib/api.js:82]: Don't do unrelated changes.

File: source/lib/badge.js [hunk 4]
```diff
@@ -24,6 +24,10 @@ const getErrorData = error => {
 	return {symbol, title};
 };
 
+export const getCurrentCount = () => {
```
Comment (sindresorhus) [inline source/lib/badge.js:27]: `export const getCurrentCount = () => browser.browserAction.getBadgeText({});`

File: source/manifest.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-	"name": "Notifier for GitHub",
+	"name": "Notifier for GitHub Custom",
```
Comment (sindresorhus) [inline source/manifest.json:2]: ?

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: source/lib/api.js [hunk 6]
```diff
@@ -90,7 +91,16 @@ export const getNotifications = async maxItems => {
...
-	const {headers, json: notifications} = await getNotificationResponse(1);
+	const {headers, json: notifications} = await getNotificationResponse(100);
+
+	// store notification urls
+	for (let n of notifications.values()) {
+		var pullIdParts = n.subject.url.split('/').slice(-4)
+		pullIdParts.splice(2, 1, 'pull')  // hack to get url to match the links clicked in /notifications
+		let pullId = pullIdParts.join('/')
+		console.log(`pushing to local storage: ${pullId}`)
+		localStore.set(pullId, true)
```
Comment (sindresorhus) [inline source/lib/api.js:102]: Same as https://github.com/sindresorhus/notifier-for-github/pull/164#discussion_r214484824

**Comment by theoschao** [comment]
apologies @sindresorhus changes here were intended for my fork only. thanks for the feedback though!

**Comment by theoschao** [comment]
that being said, this was an attempt to look into the feasibility of the proposed solution mentioned in [issue#106](https://github.com/sindresorhus/notifier-for-github/issues/106).  the notifications api unfortunately doesn't return the exact urls that are linked in the `/notifications` page, hence the less ideal url parsing
